### ISSUE TITLE
[T049] PCM classification service

### ISSUE THREAD

**Issue by wunderkennd** [wunderkennd/kaizen-web#84]
# Task T049: PCM classification model

## 📋 Overview
**Category**: Core Services
**Component**: PCM Classifier
**File Path**: `services/pcm-classifier/model/classifier.py`
**Dependencies**: T007
**Priority**: P1 - High
**Effort**: 13 story points

## 👤 User Story
As a **data scientist**, I want an ML model that accurately classifies users into PCM stages based on their behavior.

## ✅ Acceptance Criteria
- [ ] Multi-class classification (4 stages)
- [ ] >90% accuracy on test set
- [ ] Feature engineering pipeline
- [ ] Model versioning
- [ ] A/B test support
- [ ] Explainability
- [ ] Performance: <100ms inference

## 🔧 Technical Context
**Model**: XGBoost or Neural Network
**Features**: Behavior aggregates
**Training**: Offline with MLflow
**Serving**: ONNX or TensorFlow Lite

## 💡 Implementation Notes
```python
class PCMClassifier:
    def __init__(self, model_path: str):
        self.model = self._load_model(model_path)
        self.feature_extractor = FeatureExtractor()
        
    def predict(self, behavior: UserBehavior) -> PCMPrediction:
        features = self.feature_extractor.extract(behavior)
        probs = self.model.predict_proba(features)
        return PCMPrediction(
            stage=self._get_stage(probs),
            confidence=float(probs.max()),
            probabilities=probs.tolist()
        )
```

## 📚 Resources
- [XGBoost Docs](https://xgboost.readthedocs.io/)
- [Model Interpretability](https://github.com/slundberg/shap)

---
*KAIZEN Adaptive Platform - Core Service*
